### PR TITLE
feat(lingbotvlav2): support lingbot-vla-v2 model

### PR DESCRIPTION
**Pull request by XuS1994**
### Description

Add LingBot-VLA 2.0 support for PPO training and standalone evaluation on RoboTwin.

- Register `lingbotvlav2` through the existing model factory and `BasePolicy` interface.
- Implement observation/action adaptation, flow-SDE sampling, likelihood replay, and a value head. Train the action expert and value head with FSDP2 while freezing Qwen3-VL.
- Distinguish the 55-dimensional latent likelihood space from the 14-dimensional robot action space, and validate rollout/actor likelihood agreement before PPO updates.
- Install the external `lingbotvla` package from a pinned upstream revision with a versioned compatibility patch, file-hash checks, and retained license notices.
- Add training/evaluation configs, an optional GPU smoke-test CI job, a Docker target, regression tests, and English/Chinese documentation.

### Motivation and Context

RLinf's existing LingBot-VLA integration targets V1. V2 requires different model construction, preprocessing, latent-action handling, and differentiable MoE execution for PPO.

This change follows RLinf's existing adapter-plus-external-model approach and reuses its workers, FSDP2 training, checkpointing, and weight synchronization.

### How has this been tested?

**Unit tests and static checks after rebasing**

- `python -m pytest tests/unit_tests/test_models.py tests/unit_tests/test_utils.py -q`
  Result: **105 passed, 1 skipped**.
- Ruff lint/format, shell syntax, and Python/YAML parsing checks passed.
- English and Chinese documentation build checks passed.
- Confirmed that both PI0-FAST and LingBot-VLA V2 installation and CI entries are preserved.

**GPU integration testing before the final cleanup and rebase**

Environment: `5880_17`, four 48 GB GPUs, Python 3.10, PyTorch 2.9.0+cu126, Transformers 4.57.6, and FlashAttention 2.8.3.post1.

- Completed two PPO iterations on RoboTwin `click_bell` and saved a checkpoint.
- Resumed training from that checkpoint and verified optimizer-state restoration and subsequent parameter updates.
- Observed zero rollout/actor joint log-probability gap before each update, with finite losses and gradient norms.
- Verified updates to sampled action/state projection weights and the value head; the sampled frozen VLM parameter remained unchanged.
- Loaded the trained checkpoint for standalone evaluation over two environments with 100 simulator steps each.

### Additional information (optional, e.g., figures and logs):

- Upstream model source is pinned to `ecca77bb259b9592d5fc0eb2b4972d4a236ed2c8`.
- V1 and V2 require separate environments because both import as `lingbotvla`.
- GPU tests validate the training/resume/evaluation workflow; they do not establish convergence or benchmark parity.
- GPU PPO was not rerun after the final rebase. Full repository tests, clean installation, Docker builds, and upstream deployment-format export remain unvalidated.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] I have signed off my commits in accordance with the [DCO](https://github.com/RLinf/RLinf/blob/main/DCO).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
+LingBot-VLA 2.0 的 PPO 训练
```
Comment (Lin-xs) [inline docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst:1]: Please refine the documents and remove all unnecessary explanatory text.

File: docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+===========================
+
+使用 PPO 和 GAE 在 RoboTwin ``click_bell`` 上微调 LingBot-VLA 2.0，再评估保存的 policy。RLinf 通过 FSDP2 训练 action expert 和 value head，冻结 Qwen3-VL。默认训练配置使用 4 张 48 GB GPU。模型包独立安装，接入边界见 :doc:`../../extending/model_adaptation`。
+
+安装依赖
+--------
+
+准备包含 RLinf ``robotwin.envs.vector_env`` 接口的 RoboTwin 源码。安装器会固定官方 V2 源码版本，并应用 ``requirements/embodied/models/lingbotvlav2/`` 下的兼容补丁，补齐 checkpoint 元数据、数据模块延迟导入，以及概率重算和激活检查点所需的 MoE 接口。请与 :doc:`LingBot-VLA 1.0 <lingbotvla>` 使用独立 venv 和源码目录：两个包都叫 ``lingbotvla``。
```
Comment (Lin-xs) [inline docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst:9]: This paragraph can be removed. Please reorganize the rest of this section to match the structure of the “lingbot-vla 1.0” section, and rename “lingbot-vla” to “lingbotvla 1.0”.

File: docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+.. code-block:: bash
+
+   docker buildx build -f docker/Dockerfile \
+     --build-arg BUILD_TARGET=embodied-robotwin-lingbotvlav2 \
+     -t rlinf:robotwin-lingbotvlav2 .
+
+模型与资源
+----------
+
+下载官方 `V2 RoboTwin 发布模型 <https://huggingface.co/robbyant/lingbot-vla-v2-6b-robotwin>`_ 和 `Qwen3-VL-4B-Instruct <https://huggingface.co/Qwen/Qwen3-VL-4B-Instruct>`_。保留完整的 V2 发布目录：checkpoint config 只有模型类别标记，严格加载权重还需要上层的 ``lingbotvla_cli.yaml`` 和 Qwen metadata。
```
Comment (Lin-xs) [inline docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst:32]: Please provide the installation shell script.

File: docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+   export LINGBOT_VLA_V2_CKPT=/path/to/lingbot-vla-v2-6b-robotwin/checkpoints/global_step_50000/hf_ckpt
+   export LINGBOT_VLA_V2_TRAIN_CONFIG=/path/to/lingbot-vla-v2-6b-robotwin/lingbotvla_cli.yaml
+   export QWEN3_VL_PATH=/path/to/Qwen3-VL-4B-Instruct
+   # RoboTwin 要求填写包含 assets/ 的根目录。
+   export ROBOTWIN_ASSETS_PATH="$ROBOTWIN_PATH"
+   export REPO_PATH=$(pwd)
+   export PYTHONPATH="$REPO_PATH:$LINGBOT_VLA_V2_PATH:$ROBOTWIN_PATH"
+
+源码和两个模型卡均声明 Apache-2.0；兼容补丁保留原始声明并附带上游许可证。2026-09-22 检查时，这两个公开模型仓库均不需要 gated-access 审批。
```
Comment (Lin-xs) [inline docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst:44]: Please delete.

File: docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+PPO 对完整 flow-SDE latent 轨迹计算概率，保持 ``algorithm.logprob_type=chunk_level``、``algorithm.reward_type=chunk_level`` 和 ``logprob_dim=55``。BF16 权重下，记录的 latent 和概率计算仍使用 FP32，因此必须设置 ``actor.fsdp_config.mixed_precision.cast_forward_inputs=false``。``forward_micro_batch_size=1`` 保持 MoE batch shape 一致；完整轨迹概率误差超过 ``logprob_replay_atol`` 时停止更新。
+
+Smoke 在两轮更新后保存 checkpoint。使用 ``runner.resume_dir`` 续训，并保留原始 ``actor.model.model_path`` 以构造架构。检查通过后启动完整配置：
+
+.. code-block:: bash
+
+   bash examples/embodiment/run_embodiment.sh robotwin_click_bell_ppo_lingbotvlav2 ALOHA
+
+在 TensorBoard 检查 ``train/actor/total_loss``、``train/actor/grad_norm``、``train/actor/rollout_train_joint_logprob_gap_max`` 和环境成功率。仓库变量 ``LINGBOT_VLA_V2_CI=true`` 可启用 CI 作业，需要在 ``/workspace/dataset`` 预置权重和 RoboTwin 资产，并提供 4 张 GPU。全新安装和 Docker 构建需独立验证，不能用既有环境的运行结果替代。
```
Comment (Lin-xs) [inline docs/source-zh/rst_source/examples/embodied/lingbotvlav2.rst:66]: Please delete description about CI from user documents.

File: docs/source-zh/rst_source/extending/model_adaptation.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
+模型适配方案
```
Comment (Lin-xs) [inline docs/source-zh/rst_source/extending/model_adaptation.rst:1]: Please delete this page.

File: examples/embodiment/config/model/lingbotvlav2.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+model_type: lingbotvlav2
+model_path: ${oc.env:LINGBOT_VLA_V2_CKPT}
+tokenizer_path: ${oc.env:QWEN3_VL_PATH}
```
Comment (Lin-xs) [inline examples/embodiment/config/model/lingbotvlav2.yaml:3]: Please leave this field empty and instruct users to fill in the YAML fields manually, rather than passing them via environment variables.

File: requirements/embodied/models/lingbotvlav2/LICENSE [hunk 8]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (Lin-xs) [inline requirements/embodied/models/lingbotvlav2/LICENSE:1]: RLinf also follows the Apache 2.0 license. Can we delete this license file here?

File: requirements/embodied/models/lingbotvlav2/README.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# LingBot-VLA 2.0 dependency
```
Comment (Lin-xs) [inline requirements/embodied/models/lingbotvlav2/README.md:1]: This markdown file looks redundant given the existing docs. Can we drop it?

File: rlinf/models/embodiment/lingbotvlav2/lingbotvlav2_action_model.py [hunk 10]
```diff
@@ -0,0 +1,487 @@
...
+            LingbotVlaV2Policy,
+        )
+        from lingbotvla.models.vla.lingbot_vla.qwen2_action_expert import (
+            apply_lingbot_qwen2_patch,
+        )
+        from lingbotvla.models.vla.lingbot_vla.qwen3vl_in_vla import (
+            apply_lingbot_qwen3_vl_patch,
+        )
+        from safetensors.torch import load_model
+        from transformers.modeling_utils import load_sharded_checkpoint
```
Comment (Lin-xs) [inline rlinf/models/embodiment/lingbotvlav2/lingbotvlav2_action_model.py:110]: Could these imports be moved to the top of the file?

File: rlinf/models/embodiment/lingbotvlav2/lingbotvlav2_action_model.py [hunk 11]
```diff
@@ -0,0 +1,487 @@
...
+            raise ValueError("V2 flow-SDE requires num_steps >= 2 and noise_level > 0")
+        if config.rl_trainable_scope != "action_expert":
+            raise ValueError("V2 currently supports rl_trainable_scope=action_expert")
+        if config.noise_method != "flow_sde" or not config.joint_logprob:
+            raise ValueError("V2 PPO requires flow_sde and joint_logprob=true")
+        if config.is_lora:
+            raise ValueError("V2 LoRA is not supported")
+        if config.lingbotvlav2.data_parallel_backend != "fsdp2":
+            raise ValueError("V2 currently requires the FSDP2 training backend")
+        if torch.distributed.is_initialized():
```
Comment (Lin-xs) [inline rlinf/models/embodiment/lingbotvlav2/lingbotvlav2_action_model.py:129]: Is `torch.distributed.is_initialized() == False` expected here? If not, please assert it instead.

File: requirements/install.sh [hunk 12]
```diff
@@ -2635,6 +2635,42 @@ install_lingbot_vla_model() {
...
+    install_common_embodied_deps
+    # Pin the official source and apply the reviewed RLinf compatibility patch.
+    # Keep V1 and V2 in separate venvs: both distributions import as lingbotvla.
+    local vla2_path
+    local prepare_args=()
+    if [ -z "${LINGBOT_VLA_V2_PATH:-}" ] || [ ! -d "$LINGBOT_VLA_V2_PATH" ]; then
+        prepare_args+=(--managed)
+    fi
+    vla2_path=$(clone_or_reuse_repo LINGBOT_VLA_V2_PATH "$VENV_DIR/lingbot-vla-v2" "${GITHUB_PREFIX}https://github.com/robbyant/lingbot-vla-v2.git")
+    python "$SCRIPT_DIR/embodied/prepare_model_source.py" \
```
Comment (Lin-xs) [inline requirements/install.sh:2657]: Could we clone a specific commit instead, so we don’t have to run a Python script?
